### PR TITLE
Added proper find in document implementation

### LINKED ISSUE #1878: scout it does not work with multiline selections
```
air
   bat
```

`"scout two tokens air"` only searches for "air". vscode can search for multiline queries.

Haven't investigated at all, heads down on other stuff.

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
Learned of the existence of vscode command `editor.actions.findWithArgs`
https://github.com/microsoft/vscode/issues/200453

Exists in v1.66
https://github.com/microsoft/vscode/blob/release/1.66/src/vs/editor/contrib/find/browser/findModel.ts#L58

Fixes #1878

## Checklist

- [x] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [-] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [x] I have not broken the cheatsheet

### CODE REVIEW

File: packages/common/src/ide/types/ide.types.ts [hunk 1]
```diff
@@ -94,6 +94,12 @@ export interface IDE {
     listener: (event: TextDocumentChangeEvent) => void,
   ): Disposable;
 
+  /**
+   * Find occurrences of query string in the active document.
+   * @param query The string query to search for
+   */
+  findInDocument(query: string): Promise<void>;
+
```
Comment (pokey) [inline packages/common/src/ide/types/ide.types.ts:102]: Tempted to have the abstraction accept a text editor as input 🤔

File: packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/actions/scoutAir.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    - anchor: {line: 0, character: 0}
+      active: {line: 0, character: 0}
+  thatMark:
+    - type: UntypedTarget
+      contentRange:
+        start: {line: 0, character: 0}
+        end: {line: 0, character: 3}
+      isReversed: false
+      hasExplicitRange: false
+returnValue: [aaa]
```
Comment (pokey) [inline packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/actions/scoutAir.yml:34]: I don't see where the action returns anything 🤔

**Review by pokey — COMMENTED** [review]
Nice!! Left a couple minor comments

File: packages/common/src/ide/types/ide.types.ts (hunk 1 above)
Comment (AndreasArvidsson) [reply packages/common/src/ide/types/ide.types.ts:102]: Oh to open find widget in arbitrary splits?

Comment (pokey) [reply packages/common/src/ide/types/ide.types.ts:102]: Yeah, just feels like possibly a better abstraction. I don't feel strongly tho

File: packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/actions/scoutAir.yml (hunk 2 above)
Comment (AndreasArvidsson) [reply packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/actions/scoutAir.yml:34]: Hmm yeah this doesn't look right. Wrong action id even :p

File: packages/common/src/ide/types/ide.types.ts (hunk 1 above)
Comment (AndreasArvidsson) [reply packages/common/src/ide/types/ide.types.ts:102]: I agree.

Comment (pokey) [reply packages/common/src/ide/types/ide.types.ts:102]: I guess prob will be optional anyway so could possibly leave it off for now until we need it

Comment (AndreasArvidsson) [reply packages/common/src/ide/types/ide.types.ts:102]: The question is if you target something in a split do you want to search that document or the active?

File: packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/actions/scoutAir.yml (hunk 2 above)
Comment (AndreasArvidsson) [reply packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/actions/scoutAir.yml]: Tests updated

File: packages/common/src/ide/types/ide.types.ts (hunk 1 above)
Comment (AndreasArvidsson) [reply packages/common/src/ide/types/ide.types.ts:102]: If we pass the editor as an argument we can change this behavior in the future without changing the api. If we don't pass the editor we have no control over this.

Comment (pokey) [reply packages/common/src/ide/types/ide.types.ts:102]: Yeah I'm happy for us to add it. Should be trivial, no?

Fwiw I'd lean towards searching in document containing cursor, but I could see either way

Comment (AndreasArvidsson) [reply packages/common/src/ide/types/ide.types.ts]: Fixed. For now we default to the active editor, but in the future we could change this behavior easily.

**Review by pokey — APPROVED** [review]
Looks good!

**Pull request merged**